src/main/java/org/folio/edge/core/utils/test/TestUtils.java

```diff
@@ -1,60 +1,132 @@
-      Throwable t, Runnable func) {
-    Appender appender = mock(Appender.class);
+  static void getPortReset() {
+    cyclicPort.set(PORT_MAX);
+  }
+
+  /**
+   * Check a local TCP port.
+   * @param port  the TCP port number, must be from 1 ... 65535
+   * @return true if the port is free (unused), false if the port is already in use
+   */
+  static boolean isLocalPortFree(int port) {
+      try {
+          new ServerSocket(port).close();
+          return true;
+      } catch (IOException e) {
+          return false;
+      }
+  }
 
-    assertNotNull(logger);
-    assertNotNull(func);
+  /**
+   * Unit test for logged content.
+   *
+   * @param logger that is used in test code
+   * @param minTimes minimum number of log events
+   * @param maxTimes maximum number of log events
+   * @param logLevel level to test against (not working, so not checked)
+   * @param expectedMsg expected message for first message
+   * @param t expected Throwable (not working, pass null always)
+   * @param func function to execute which presumably logs
+   */
+  public static void assertLogMessage(Logger logger, int minTimes, int maxTimes, Level logLevel,
+                                      String expectedMsg, Throwable t, Runnable func) {
+    assertLogMessage((org.apache.logging.log4j.core.Logger) logger, minTimes, maxTimes, logLevel, expectedMsg, t, func);
+  }
 
-    try {
-      logger.addAppender(appender);
-      ArgumentCaptor<LoggingEvent> argument = ArgumentCaptor.forClass(LoggingEvent.class);
+  private static void assertLogMessage(org.apache.logging.log4j.core.Logger logger, int minTimes, int maxTimes, Level logLevel,
+                                       String expectedMsg, Throwable t, Runnable func) {
 
-      func.run();
+    AppenderCapture appender = new AppenderCapture();
 
-      verify(appender, atLeast(minTimes)).doAppend(argument.capture());
-      verify(appender, atMost(maxTimes)).doAppend(argument.capture());
+    logger.addAppender(appender);
+    func.run();
+    appender.stop();
+    logger.removeAppender(appender);
 
-      if (logLevel != null)
-        assertEquals(logLevel, argument.getValue().getLevel());
+    Assert.assertTrue(appender.events.size() >= minTimes);
+    Assert.assertTrue(appender.events.size() <= maxTimes);
 
-      if (expectedMsg != null)
-        assertEquals(expectedMsg, argument.getValue().getMessage());
+    Assert.assertNull(t);
 
-      if (t != null) {
-        assertNotNull(argument.getValue().getThrowableInformation());
-        assertEquals(t, argument.getValue().getThrowableInformation().getThrowable());
-      }
-    } finally {
-      logger.removeAppender(appender);
+    if (logLevel != null) {
+      Assert.assertTrue(!appender.events.isEmpty());
+      Level gotLevel = appender.events.get(0).getLevel();
+      Assert.assertTrue(gotLevel == Level.OFF || gotLevel == logLevel);
```

Comment:
I would expect this to fail if the log level doesn't match what was provided.  Why the special case for `Level.OFF`?

Reply:
The implementation is BROKEN!! The system always returns Level.OFF. I've spent 4 hours to figure out why.. Open to suggestions.. However, this utility is ONLY used in edge-common.

Reply:
interesting.  I don't have much log4j2 experience.  @mreno-EBSCO ?

Reply:
I had started to upgrade to log4j2 in my local edge-common repo and ended up removing the ability to assert log messages as I wasn't making much progress. I also was motivated to remove compile time dependencies on test libraries, which end up as required dependencies in the edge modules, so maybe I didn't try hard enough. 😄

Reply:
How about we just leave it for now, but remove it for version 4.

Alternatively, I'm happy to remove it in this branch. I don't know of ANY modules using that feature from mod-edge.

Reply:
It's probably safer to remove it in v4.  That get's my vote.